sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/reactive/PanacheEndpoint.java

```diff
@@ -0,0 +1,135 @@
+                        : Response.ok(book.getISBN()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @PUT
+    @Path("isbn/{id}/{isbn}")
+    public Uni<Response> changeISBN(Integer id, Long isbn) {
+        return Book.byId(id)
+                .flatMap(book -> {
+                    book.setISBN(isbn);
+                    return book.persistAndFlush();
+                })
+                .map(book -> book == null
+                        ? Response.status(Response.Status.NOT_FOUND)
+                        : Response.ok())
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @PUT
+    @Path("books/{author}/{name}")
+    public Uni<Response> createBook(Integer author, String name) {
+        return Book.create(author, name)
+                .map(nothing -> Response.status(Response.Status.CREATED))
+                .onFailure().recoverWithItem(error -> Response.status(Response.Status.BAD_REQUEST).entity(error.getMessage()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @GET
+    @Path("books/author/{name}")
+    public Multi<String> search(String name) {
+        return authors.findByName(name)
+                .flatMap(Author::getBooks)
+                .map(Book::getTitle);
+    }
+
+    @GET
+    @Path("author")
+    // TODO: Return Json, when this will be fixed: https://github.com/quarkusio/quarkus/issues/18043
+    public Multi<String> author() {
+        return authors.streamAll()
+                .map(author -> author.getId() + " " + author.getName());
+    }
+
+    @GET
+    @Path("author/{id}")
+    public Uni<Response> author(Integer id) {
+        return authors.findById(id)
+                .map(author -> author == null
+                        ? Response.status(Response.Status.NOT_FOUND)
+                        : Response.ok(author.getName()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @POST
+    @Path("author/{name}")
+    public Uni<Response> createAuthor(String name) {
+        return authors.create(name)
+                .map(ignored -> Response.status(Response.Status.CREATED))
+                .onFailure().recoverWithItem(throwable -> {
+                    return Response.status(Response.Status.BAD_REQUEST).entity(throwable.getMessage());
+                })
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @DELETE
+    @Path("author/{id}")
+    public Uni<Response> deleteAuthor(Integer id) {
+        return authors.deleteById(id)
+                .call(isDeleted -> authors.flush())
+                .map(isDeleted -> isDeleted
+                        ? Response.status(Response.Status.NO_CONTENT)
+                        : Response.status(Response.Status.NOT_FOUND))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @GET
+    @Path("dto")
+    public Uni<Response> dtoQuery() {
+        return Book.find("author", 4).project(BookDescription.class).list()
```

Comment:
We need to avoid the use of "magic numbers" in the code. Find more appropriate filter parameter

Reply:
Added id to url